src/cmap/connection.ts

```diff
@@ -488,12 +511,19 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       return;
     }
 
-    this[kStream].end(() => {
+    if (this[kStream].destroyed) {
```

Comment:
Does it make sense to use [`.writableEnded`](https://nodejs.org/api/stream.html#writablewritableended) here? For writable streams there is a window of time where `.end()` has been called but `close` has not been emitted/`.destroyed` is still false.